Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -267,8 +343,18 @@ public class HTTPClient {
     ///     - request: HTTP request to execute.
     ///     - deadline: Point in time by which the request must complete.
     public func execute(request: Request, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
+        return self.execute(request: request, deadline: deadline, logger: HTTPClient.loggingDisabled)
+    }
+
+    /// Execute arbitrary HTTP request using specified URL.
+    ///
+    /// - parameters:
+    ///     - request: HTTP request to execute.
+    ///     - deadline: Point in time by which the request must complete.
+    ///     - logger: The logger to use for this request.
+    public func execute(request: Request, deadline: NIODeadline? = nil, logger: Logger) -> EventLoopFuture<Response> {
```

Comment:
Curious, why as overloads and not `logger: Logger = HTTPClient.loggingDisabled`?

Reply:
@ktoso breaks the public API if you add parameters. Imagine a user had written

```
func myDecoratedExecute(request: Request, _ execute: (Request, NIODeadline?)) -> ELF<Response>) -> ELF<Response> {
    print("start")
    return execute(request, .seconds(5))
}
```

and called it with 

```
myDecoratedExecute(request, self.myHTTPClient.execute)
```

this would now break if we added a new parameter.

Reply:
Ok I see, in the implicit case because someone took that closure so the signature is locked in... Okey 👍